### PR TITLE
Redesign cancel match modal, Added modal template

### LINKED ISSUE #13: Redesign remove match modal
The remove match modal should be redesigned to match the other modals on the dashboard page. A design proposal has already been made in XD.

### PR DESCRIPTION
**Pull request by realmayus**
This PR adds a template for creating consistent modals with a customizable accent color and icon.
Also, it re-implemented the match cancelling modal.

Fixes #13

### CODE REVIEW

File: src/components/Modals/CancelMatchModal.tsx [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+                  selected={newMatchWanted}
+                  onSelect={setNewMatchWanted}
+                  label="Ja"
+                />
+                <DialogModalBase.CheckboxSingle
+                  value={false}
+                  selected={newMatchWanted}
+                  onSelect={setNewMatchWanted}
+                  label="Nein"
+                />
```
Comment (lsc0498) [inline src/components/Modals/CancelMatchModal.tsx:236]: Maybe a confirm button here would be a good idea. Or alternatively buttons for "Ja"or "Nein". From my user perspective I'm not convinced before leaving the modal that my choice will persist.
Bildschirmfoto 2020-12-05 um 03 59 54

**Review by lsc0498 — COMMENTED** [review]
Looks good to me. Great job!👍

**Review by geroembser — COMMENTED** [review]
I really like the styling, and it looks like good code! Well done!
However I would like the following things to be addressed before merging:
- [ ] It would be good, if the match dissolution modal for project coaching will also use your newly implemented modal.
- [ ] Support for mobile devices should be improved. It seems to me that scrolling on smaller screens does not work as expected (e.g. on iPhone SE).

File: src/components/Modals/CancelMatchModal.tsx (hunk 1 above)
Comment (realmayus) [reply src/components/Modals/CancelMatchModal.tsx:252]: Fixed in a1002b88b4f361ff88fa73cfd0284bbee2015483
Screen Shot 2020-12-05 at 11 06 41

**Review by geroembser — APPROVED** [review]

**Review by lsc0498 — APPROVED** [review]

**Pull request merged**